src/openfermion/utils/_sparse_tools.py

```diff
@@ -250,6 +250,68 @@ def jw_number_restrict_operator(operator, n_electrons, n_qubits=None):
     return operator[numpy.ix_(select_indices, select_indices)]
 
 
+def jw_get_ground_states_by_particle_number(sparse_operator, particle_number):
+    """For a Jordan-Wigner encoded operator, compute the lowest eigenvalue and
+    eigenstates at a particular particle number. The operator must conserve
+    particle number.
+
+    Args:
+        sparse_operator: A Jordan-Wigner encoded sparse operator.
```

Comment:
While we are not always consistent with this, we try to indicate the type of all args / returns by specifying it in parenthesis right after the name. For instance, you would write:
 `particle_number(int): The particle number at which to compute ground states`

This is perhaps even more important for something like "ground_states". You should specify both that the output is a list but also what are the elements of the list (in this case, a numpy array).